data-source-manager/src/main/java/com/odysseusinc/datasourcemanager/jdbc/DataSourceJdbcExecutor.java

```diff
@@ -0,0 +1,63 @@
+package com.odysseusinc.datasourcemanager.jdbc;
+
+import com.odysseusinc.arachne.commons.types.DBMSType;
+import com.odysseusinc.arachne.execution_engine_common.api.v1.dto.DataSourceUnsecuredDTO;
+import com.odysseusinc.arachne.execution_engine_common.util.BigQueryUtils;
+import com.odysseusinc.datasourcemanager.krblogin.KerberosService;
+import com.odysseusinc.datasourcemanager.krblogin.RuntimeServiceMode;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.util.Objects;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+
+public class DataSourceJdbcExecutor {
+
+	private final KerberosService kerberosService;
+
+	public DataSourceJdbcExecutor(KerberosService kerberosService) {
+
+		this.kerberosService = kerberosService;
+	}
+
+	public <T> T executeOnSource(DataSourceUnsecuredDTO dataSourceData, JdbcTemplateConsumer<T> consumer) throws IOException {
+
+		if (Objects.isNull(consumer)) {
+			throw new IllegalArgumentException("consumer is required");
+		}
+
+		File tempDir = Files.createTempDirectory("gis").toFile();
+
+		// BigQuery
+		if (Objects.equals(DBMSType.BIGQUERY, dataSourceData.getType()) && Objects.nonNull(dataSourceData.getKeyfile())) {
```

Comment:
I believe the authentication related parts should be extracted into separate methods so that we could re-use them in https://github.com/OHDSI/ArachneExecutionEngine/blob/develop/src/main/java/com/odysseusinc/arachne/executionengine/service/impl/AnalysisServiceImpl.java#L113 . Also, get a look at the `resolveBqAuth` and `resolveKerberosAuth` in the `AnalysisServiceImpl` - may be we can introduce a mechanism of Data source auth resolvers?